Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+
+    TS_ASSERT(compInfo.hasSample());
+    TS_ASSERT(tester.parentNXgroupHasChildNXgroup(NX_ENTRY, NX_SAMPLE));
+  }
+
+  /*
+====================================================================
+
+NEXUS TRANSFOMATIONS TESTS
+
+DESCRIPTION:
+
+The following tests document that saveInstrument will find and write detectors
+and other Instrument components to file in Nexus format, and where there exists
+transformations in ComponentInfo and DetectorInfo, SaveInstrument will generate
+'NXtransformations' groups to contain the corresponding component
+rotations/translations, and pixel offsets in any 'NXdetector' or 'NXmonitor'
+found in the Instrument cache.
+
+LIST(S):
+
+* test when NXdetector groups have NXtransformations, 'transformation_type' is
+specified for all
+* test when NXmonitor groups have NXtransformations, 'transformation_type' is
+specified for all
+*  test when NXsource group has NXtransformations, 'transformation_type' is
+specified
+* test x/y/z pixel offset in file is relative position from bank without
+rotation
+
+* test rotation of NXdetector written to file is same as in component info.
+* test rotation of NXmonitor written to file is same as in component info.
+* test rotation of source written to file is same as in component info.
+
+LOCATION TESTS:
+* test NXmonitor location not written when is at origin
+* test NXdetector location not written when is at origin
+* test NXsource location not written when is at origin
+
+ROTATION TESTS
+* test NXdetector rotation not written when is zero
+* test NXmonitor rotation not written when is zero
+* test NXsource rotation not written when is zero
+
+====================================================================
+*/
+
+  void
+  test_when_nx_detector_groups_have_nx_transformations_transformation_type_is_specified_for_all() {
+
+    ScopedFileHandle fileResource(
+        "check_nxdetector_groups_have_transformation_types_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    auto instrument =
+        ComponentCreationHelper::createTestInstrumentRectangular2(2, 2);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+
+    // saveinstrument
+    NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                      DEFAULT_ROOT_PATH);
+    auto &compInfo = (*instr.first);
+
+    auto instrName = compInfo.name(compInfo.root());
+
+    HDF5FileTestUtility tester(destinationFile);
+
+    for (size_t i = compInfo.root() - 1; i > 0; --i) {
+
+      if (Mantid::Geometry::ComponentInfoBankHelpers::isAnyBank(compInfo, i)) {
+
+        fullH5Path path = {DEFAULT_ROOT_PATH, instrName, compInfo.name(i),
+                           TRANSFORMATIONS};
+        bool hasNXTransformation =
+            tester.hasAttributeInGroup(NX_CLASS, NX_TRANSFORMATIONS, path);
+
+        // TODO: having such a group may be optional.
+        TS_ASSERT(hasNXTransformation);
```

Comment:
Would prefer that you refactored to do the `TS_ASSERT` and method calls in one line. i.e.

`TS_ASSERT(tester.hasAttributeInGroup(NX_CLASS, NX_TRANSFORMATIONS, path))` 
Apply this pattern across your test file.